### PR TITLE
Added class path entry allocation.

### PR DESCRIPTION
**Pull request by akshayben**
- class path entry allocation found in jvminit initialize Class path
- error during warm run when classpath find is called
- class path entry extra info allocation also done
- free method in jvmfree changed for class path entries

Signed-off-by: akshayben akshayben@ibm.com

### CODE REVIEW

**Comment by akshayben** [comment]
Class path entry does not seem like its allocated into the heap. We allocated in cphelp.c but jvminit.c also allocates cpentries.

**Comment by akshayben** [comment]
image

**Comment by akshayben** [comment]
The trace shows error at classsupport.c:1198 which is not entered into during a normal run without ram saving/restoring. class path entries of system class loader is therefore not correctly allocated. In cphelp.c `addJarToSystemClassLoaderClassPathEntries` is never called when running `java -version`

**Review by Ershany-IBM — APPROVED** [review]

**Review by VinoshanT — APPROVED** [review]

File: runtime/vm/jvminit.c [hunk 1]
```diff
@@ -1402,7 +1410,12 @@ initializeClassPathEntry (J9JavaVM * javaVM, J9ClassPathEntry *cpEntry)
 		VMIZipFile *zipFile = NULL;
 
 		cpEntry->extraInfo = NULL;
-		zipFile = j9mem_allocate_memory((UDATA) sizeof(*zipFile), J9MEM_CATEGORY_CLASSES);
+		if (IS_COLD_RUN(javaVM)) {
```
Comment (akshayben) [inline runtime/vm/jvminit.c]: zip file (extraInfo) shouldn't be cached. Fixup of class path entries needed.

**Pull request merged**